### PR TITLE
NNVP-15 initial

### PR DESCRIPTION
**Pull request by eryk7b**

### CODE REVIEW

File: src/main/java/com/theglue/nnvp/NnvpApplication.java [hunk 1]
```diff
@@ -19,21 +26,21 @@ public static void main(String[] args) {
     SpringApplication.run(NnvpApplication.class, args);
 
     System.out.println(
```
Comment (mateusztheglue) [inline src/main/java/com/theglue/nnvp/NnvpApplication.java]: seems we still have some different formatting settings

File: src/main/java/com/theglue/nnvp/controller/VehiclesRestController.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@Log4j2
+@RestController
+public class VehiclesRestController {
```
Comment (mateusztheglue) [inline src/main/java/com/theglue/nnvp/controller/VehiclesRestController.java]: lets use singular form Vehicle
that is a general remark

**Comment by theglue-cicd** [comment]
:star: PR built and available in a preview environment **the-glue-nnvp-backend-pr-2** [here](http://nnvp-backend.jx-the-glue-nnvp-backend-pr-2.cluster.development.nnvp.thegluesandbox.com)

File: src/main/java/com/theglue/nnvp/controller/dto/VehicleDTO.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package com.theglue.nnvp.controller.dto;
+
+import lombok.*;
```
Comment (mateusztheglue) [inline src/main/java/com/theglue/nnvp/controller/dto/VehicleDTO.java]: import *

File: src/main/java/com/theglue/nnvp/controller/dto/VehicleDTO.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+import javax.persistence.Column;
+import javax.persistence.Id;
+
+@Getter
+@Setter
+@With
+@AllArgsConstructor
+@NoArgsConstructor
```
Comment (mateusztheglue) [inline src/main/java/com/theglue/nnvp/controller/dto/VehicleDTO.java]: maybe let's use @Data instead?

File: src/main/java/com/theglue/nnvp/controller/dto/VehicleDTO.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+public class VehicleDTO {
+
+  @Id
+  private String id;
+
+  @Column
+  private String make;
```
Comment (mateusztheglue) [inline src/main/java/com/theglue/nnvp/controller/dto/VehicleDTO.java]: jpa annotations in DTO?

File: src/main/java/com/theglue/nnvp/entity/Vehicle.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Getter
+@Setter
+@With
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(name = "vehicle")
```
Comment (mateusztheglue) [inline src/main/java/com/theglue/nnvp/entity/Vehicle.java]: I think @Data fits here as well?
No need for @Table, let's use the default

File: src/main/resources/application.yml [hunk 7]
```diff
@@ -1,11 +1,19 @@
 spring:
   application:
     name: NNVP-backend
-
```
Comment (mateusztheglue) [inline src/main/resources/application.yml]: this should be in application-test.yml

**Comment by theglue-cicd** [comment]
:star: PR built and available in a preview environment **the-glue-nnvp-backend-pr-2** [here](http://nnvp-backend.jx-the-glue-nnvp-backend-pr-2.cluster.development.nnvp.thegluesandbox.com)

**Pull request merged**